ipyopenlayers/example.py

```diff
@@ -22,7 +23,28 @@ class TileLayer(Widget):
     _view_module = Unicode(module_name).tag(sync=True)
     _view_module_version = Unicode(module_version).tag(sync=True)
     url = Unicode('https://{a-c}.tile.openstreetmap.org/{z}/{x}/{y}.png').tag(sync=True)
-   
+
+class BaseOverlay(DOMWidget):
+   _model_name = Unicode('BaseOverlayModel').tag(sync=True)
+   _model_module = Unicode(module_name).tag(sync=True)
+   _model_module_version = Unicode(module_version).tag(sync=True)
+   _view_name = Unicode('BaseOverlayView').tag(sync=True)
+   _view_module = Unicode(module_name).tag(sync=True)
+   _view_module_version = Unicode(module_version).tag(sync=True)
+   overlay_type = Unicode().tag(sync=True)
+   position = List([0, 0]).tag(sync=True)
+
+class ImageOverlay (BaseOverlay):
+   overlay_type = Unicode('image').tag(sync=True)
+   image_url = Unicode('').tag(sync=True)
+
+class VideoOverlay (BaseOverlay):
+   overlay_type = Unicode('video').tag(sync=True)
+   video_url = Unicode('').tag(sync=True)
+
+class PopupOverlay (BaseOverlay):
+   overlay_type = Unicode('popup').tag(sync=True)
+   popup_content = Unicode('').tag(sync=True)
```

Comment:
This approach may be fine for now, but it does not scale well if you have many types of overlays. The `BaseOverlayView` implementation front-end side becomes too complicated with a big `if` condition and many different methods depending on the type.

It would be nicer to have one widget implementation per overlay type:

```suggestion
class BaseOverlay(DOMWidget):
   _model_module = Unicode(module_name).tag(sync=True)
   _model_module_version = Unicode(module_version).tag(sync=True)
   _view_module = Unicode(module_name).tag(sync=True)
   _view_module_version = Unicode(module_version).tag(sync=True)

   overlay_type = Unicode().tag(sync=True)
   position = List([0, 0]).tag(sync=True)

class ImageOverlay (BaseOverlay):
   _view_name = Unicode('ImageOverlayView').tag(sync=True)
   _model_name = Unicode('ImageOverlayModel').tag(sync=True)

   overlay_type = Unicode('image').tag(sync=True)
   image_url = Unicode('').tag(sync=True)

class VideoOverlay (BaseOverlay):
   _view_name = Unicode('VideoOverlayView').tag(sync=True)
   _model_name = Unicode('VideoOverlayModel').tag(sync=True)
   
   overlay_type = Unicode('video').tag(sync=True)
   video_url = Unicode('').tag(sync=True)

class PopupOverlay (BaseOverlay):
   _view_name = Unicode('PopupOverlayView').tag(sync=True)
   _model_name = Unicode('PopupOverlayModel').tag(sync=True)
   
   overlay_type = Unicode('popup').tag(sync=True)
   popup_content = Unicode('').tag(sync=True)
```

This means implementing an `ImageOverlayView`, `ViewOverlayView` and `PopupOverlayView` front-end side.

Let's keep this comment for a future PR and merge.